src/sdk/wire_bytes.rs

```diff
@@ -27,41 +32,62 @@ pub fn encode<T: Serialize>(payload: &T) -> TofnResult<BytesVec> {
     })
 }
 
+/// Serialize a value using bincode and log errors
 pub fn serialize<T: ?Sized>(value: &T) -> TofnResult<BytesVec>
 where
     T: serde::Serialize,
 {
-    match bincode::serialize(value) {
-        Ok(bytes) => Ok(bytes),
-        Err(err) => {
-            error!("serialization failure: {}", err.to_string());
-            Err(TofnFatal)
-        }
-    }
+    // Create serialization options for bincode.
+    // The default options don't bound pre-allocation size,
+    // use little-endian and varint encoding, and reject trailing bytes.
+    let options = DefaultOptions::new()
+        .with_limit(MAX_MSG_LEN)
+        .with_big_endian() // do not ignore extra bytes at the end of the buffer
+        .with_varint_encoding() // saves a lot of space in smaller messages
+        .reject_trailing_bytes(); // do not ignore extra bytes at the end of the buffer
+
+    options.serialize(value).map_err(|err| {
+        error!("serialization failure: {}", err.to_string());
+        TofnFatal
+    })
+}
+
+/// Deserialize bytes to a type using bincode and log errors
+pub fn deserialize<T: DeserializeOwned>(bytes: &[u8]) -> TofnResult<T> {
```

Comment:
Should we return a `bincode` result or `Option` instead of a `TofnResult`?  `TofnFatal` is intended only for fatal errors.